lib/blacklist.js

```diff
@@ -1,48 +1,59 @@
+const _ = require('lodash/fp')
+
 const db = require('./db')
 const notifierQueries = require('./notifier/queries')
 
-// Get all blacklist rows from the DB "blacklist" table that were manually inserted by the operator
 const getBlacklist = () => {
-  return db.any(`SELECT * FROM blacklist`).then(res =>
-    res.map(item => ({
-      cryptoCode: item.crypto_code,
-      address: item.address
-    }))
-  )
+  const blacklistSql = `SELECT * FROM blacklist`
+  const messagesSql = `SELECT * FROM blacklist_messages`
+  return Promise.all([db.any(blacklistSql), db.any(messagesSql)])
+    .then(([blacklist, messages]) => Promise.all([_.map(_.mapKeys(_.camelCase), blacklist), _.map(_.mapKeys(_.camelCase), messages)]))
+    .then(([blacklist, messages]) => _.map(it => ({ ...it, blacklistMessage: _.find(ite => it.blacklistMessageId === ite.id, messages) }), blacklist))
 }
 
-// Delete row from blacklist table by crypto code and address
-const deleteFromBlacklist = (cryptoCode, address) => {
-  const sql = `DELETE FROM blacklist WHERE crypto_code = $1 AND address = $2`
-  notifierQueries.clearBlacklistNotification(cryptoCode, address)
-  return db.none(sql, [cryptoCode, address])
+const deleteFromBlacklist = address => {
+  const sql = `DELETE FROM blacklist WHERE address = $1`
+  notifierQueries.clearBlacklistNotification(address)
+  return db.none(sql, [address])
 }
 
-const insertIntoBlacklist = (cryptoCode, address) => {
+const insertIntoBlacklist = address => {
   return db
     .none(
-      'INSERT INTO blacklist (crypto_code, address) VALUES ($1, $2);',
-      [cryptoCode, address]
+      'INSERT INTO blacklist (address) VALUES ($1);',
+      [address]
     )
 }
 
-function blocked (address, cryptoCode) {
-  const sql = `SELECT * FROM blacklist WHERE address = $1 AND crypto_code = $2`
-  return db.any(sql, [address, cryptoCode])
+function blocked (address) {
+  const sql = `SELECT address, content FROM blacklist b LEFT OUTER JOIN blacklist_messages bm ON bm.id = b.blacklist_message_id WHERE address = $1`
+  return db.any(sql, [address])
```

Comment:
There's an `UNIQUE` constraint on the address, so `oneOrNone()` could be more appropriate?